### DISCUSSION TITLE
Stacked mergeMode. How is the value presented?

### DISCUSSION THREAD

**Discussion by Curwe** [patrykandpatrick/vico#547] [Questions] [answered]
Hi. 

I'm on latest stable. 

We have an app that use the stacked mergeMode. Our data is composed as this and then we make a list of that.
```
data class Values(
      val total: Float, // Combined value of sold and consumed. 
      val sold: Float,
      val consumed: Float
)
```

The default logic is that the graph present just the total values. But with some logic in the vm, when a user tapps a column we replace the total value with the sold and consumed values on that same "x" position. 
Since we use mergeMode.Stack values on the same x axis should be drawn on the same x position. 

Okay so here is where the issue happen.. the sold + consumption = total. But when it's rendered, the columns Y values never seem to be equal to the total value. 
Any idea on what to do here?

**Comment by patrickmichalik** [comment, marked as answer]
Hello! `MergeMode.Stack` is sum-based. The total height of a column stack should be the same as the height of a single column whose _y_ value is the sum of the stacked columns’ _y_ values. This works as expected in our testing. Here’s a simple example. The sums are 3 + 2 = 5 and 2 + 6 = 8. The column stacks have the expected total heights.

```kt
val chartEntryModel = entryModelOf(entriesOf(3f, 2f), entriesOf(2f, 6f))
Chart(
    chart = columnChart(mergeMode = ColumnChart.MergeMode.Stack),
    model = chartEntryModel,
    startAxis = rememberStartAxis(),
    bottomAxis = rememberBottomAxis(),
)
```

<img src="https://github.com/patrykandpatrick/vico/assets/120058021/98312303-f0c8-4d40-813c-b024c0451a94" width="720" />

Thus, to determine what’s happening, we’ll need some more information. Could you please provide an example where the issue occurs?

**Reply by Curwe** [reply]
Thanks, sorry for the late replay!

I misunderstood the usage and that's clear now. 
The way I used it was all in a single list like this.

```
listOf(
     FloatEntry(x = 0, y = 10),
     FloatEntry(x = 0, y = 5),
     FloatEntry(x = 1, y = 10),
     FloatEntry(x = 2, y = 10),
     FloatEntry(x = 3, y = 10),
)
```

And I expected it to work since the x value was the same. But it just ended up using the last added at that index. 
But now I'm using two separate lists and it works a expected. Thanks

**Reply by patrickmichalik** [reply]
I’m glad to hear that. You’re welcome!
